src/dotty/tools/dotc/transform/ElimRepeated.scala

```diff
@@ -43,6 +43,8 @@ class ElimRepeated extends MiniPhaseTransform with InfoTransformer with Annotati
           paramTypes.init :+ last
         } else paramTypes
       tp.derivedMethodType(paramNames, paramTypes1, resultType1)
+    case tp: ExprType if tp.resultType.isRepeatedParam =>
+       ExprType(tp.resultType.underlyingIfRepeated(isJava = false))
```

Comment:
>Question: Did you compile StringContext with dotty, or import it with the unpickler? 

It was unpickled from Scala2.
Similar code does not typecheck with dotty: https://github.com/lampepfl/dotty/pull/768